Add support for override host field

philippschulte opened the pull request:
🎉this closes #163.

philippschulte committed:
feat(backend): add support for override_host field

The override_host field can be used to override the Host header.

fix #163

philippschulte committed:
docs(backend): add support for override_host field

fix #163

trentrosenbaum reviewed (changes requested):
From the point of view of introducing a new field, this looks ok and the field is documented in the provider and on the fastly api website.  I cannot see any documented validation and so this should be fine.

The current build is failing and so just marking it as "Request Changes" to see a green build.

philippschulte committed:
build(client): update go-fastly to v1.0.0

BREAKING CHANGE: API for BigQuery logging has changed in v1.0.0 and needs to be adjusted in the
provider.

fix #163

philippschulte committed:
fix(bigquery): API request to list all logging objects

The go-fastly client API for BigQuery has changed in v1.0.0.

philippschulte committed:
test(director): fix response timestamp fields

The go-fastly client API for Director has changed in v1.0.0.

philippschulte committed:
test(snippet): fix response timestamp fields

The go-fastly client API for VCL Snippets has changed in v1.0.0.

philippschulte commented:
@trentrosenbaum the build wasn't passing because the client hasn't been updated. The PR updates the `go-fastly` client from v0.4.3 to v1.0.0 in order to support the new `override_host` field. By doing so some fixes were necessary because v1.0.0 includes some breaking changes. Please review it again.
